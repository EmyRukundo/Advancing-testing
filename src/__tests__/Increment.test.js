import React from 'react';
import {shallow } from '../enzyme';
import Increment from '../components/Increment';


describe('App component', () => {
    it('starts with a count of 0', () => {
        const wrapper = shallow(<Increment />);
        const text = wrapper.find('p').text();
        expect(text).toEqual('Count: 0');
    });

    it('increments count by 1 when the increment button is clicked', () => {
        const wrapper = shallow(<Increment />);
        const incrementBtn = wrapper.find('button.increment');
        incrementBtn.simulate('click');
        const text = wrapper.find('p').text();
        expect(text).toEqual('Count: 0');
    });
    
    it('decrements count by 1 when the decrement button is clicked', () => {
        const wrapper = shallow(<Increment />);
        const decrementBtn = wrapper.find('button.decrement');
        decrementBtn.simulate('click');
        const text = wrapper.find('p').text();
        expect(text).toEqual('Count: -1');
    });
    it('matches snapshots ', () => {
        const wrapper = shallow(<Increment />);
        expect(wrapper).toMatchSnapshot();
    })
});