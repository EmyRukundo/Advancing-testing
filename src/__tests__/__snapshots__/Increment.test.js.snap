// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshots  1`] = `ShallowWrapper {}`;
